include/llbuild/BuildSystem/BuildValue.h

```diff
@@ -52,15 +52,21 @@ class BuildValue {
 
     /// A value for a produced output whose command failed.
     FailedInput,
+      
+    /// A value for a produced output whose command was cancelled.
+    CancelledInput,
 
     /// A value produced by a successful command.
     SuccessfulCommand,
 
     /// A value produced by a failing command.
     FailedCommand,
-      
-    /// A value produced by a command which was skipped.
-    SkippedCommand,
+
+    /// A value produced by a command which was skipped because one of its dependencies failed.
+    FailedDependencyCommand,
```

Comment:
I don't have a better idea, but I would probably pick a different name from `FailedDependencyCommand`, just because it is hard to read when juxtaposed with `FailedCommand`... what about `UnavailableCommand` (under idea it is unavailable because it couldn't run because input was missing) or any other ideas? `SkippedCommandBecauseOfPropagatedFailure` seems too long. :)